File: app/src/main/java/com/example/notes/note/papers/paper/messages/message/scraps/scrap/letters/letter/memoirs/memoir/data/NoteDataConvertor.java
package com.example.notes.note.papers.paper.messages.message.scraps.scrap.letters.letter.memoirs.memoir.data;

import com.google.firebase.Timestamp;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

public class NoteDataConvertor {
    public static class Fields {
        public final static String TITLE = "title";
        public final static String NOTE_CONTENT = "note content";
        public final static String DATE_OF_CREATION = "date of creation";
        public final static String DATE_OF_EDIT = "date of edit";
        public final static String IS_FAVORITE = "is favorite";

    }

    public static NoteData documentToNoteData(String id, Map<String, Object> doc) {
        NoteData.Builder noteBuilder = new NoteData
                .Builder()
                .setId(id)
                .setTitle((String) doc.get(Fields.TITLE))
                .setNoteContent((String) doc.get(Fields.NOTE_CONTENT))
                .setIsFavorite((boolean) doc.get(Fields.IS_FAVORITE));
                //.setDateOfCreation(((Timestamp) (doc.get(Fields.DATE_OF_CREATION))).toDate())
                //.setDateOfEdit(((Timestamp) (doc.get(Fields.DATE_OF_EDIT))).toDate());

        return noteBuilder.build();
    }

     public static Map<String, Object> noteDataToDocument( NoteData note) {
        Map<String, Object> doc = new HashMap<>();
        doc.put(Fields.TITLE, note.getTitle());
        doc.put(Fields.NOTE_CONTENT, note.getNoteContent());
        doc.put(Fields.DATE_OF_CREATION, note.getDateOfCreation());
        doc.put(Fields.DATE_OF_EDIT, note.getDateOfEdit());
        doc.put(Fields.IS_FAVORITE, note.isFavorite());

        return doc;
    }
}

File: app/src/main/java/com/example/notes/note/papers/paper/messages/message/scraps/scrap/letters/letter/memoirs/memoir/data/NoteDataConvertor.kt
package com.example.notes.note.papers.paper.messages.message.scraps.scrap.letters.letter.memoirs.memoir.data

import com.example.notes.note.papers.paper.messages.message.scraps.scrap.letters.letter.memoirs.memoir.data.NoteData
import java.util.*

object NoteDataConvertor {
    fun documentToNoteData(id: String?, doc: Map<String?, Any?>): NoteData {
        val noteBuilder = NoteData.Builder()
            .setId(id)
            .setTitle(doc[Fields.TITLE] as String?)
            .setNoteContent(doc[Fields.NOTE_CONTENT] as String?)
            .setIsFavorite(doc[Fields.IS_FAVORITE] as Boolean)
           //.setDateOfCreation(doc[Fields.DATE_OF_CREATION] as Date)
            //.setDateOfCreation(doc[Fields.DATE_OF_EDIT] as Date)

        return noteBuilder.build()
    }

    fun noteDataToDocument(note: NoteData?): Map<String, Any?> {
        val doc: MutableMap<String, Any?> = HashMap()
        if(note!=null) {
            doc[Fields.TITLE] = note.title
            doc[Fields.NOTE_CONTENT] = note.noteContent
            doc[Fields.DATE_OF_CREATION] = note.dateOfCreation
            doc[Fields.DATE_OF_EDIT] = note.dateOfEdit
            doc[Fields.IS_FAVORITE] = note.isFavorite
        }
        return doc
    }

    object Fields {
        const val TITLE = "title"
        const val NOTE_CONTENT = "note content"
        const val DATE_OF_CREATION = "date of creation"
        const val DATE_OF_EDIT = "date of edit"
        const val IS_FAVORITE = "is favorite"
    }
}